src/php/functions.php

```diff
@@ -74,6 +74,7 @@
  */
 function add_to_context( $context ) {
 	$context['menu'] = new \Timber\Menu( 'primary-menu' );
+	$context['primary_sidebar'] = Timber\Timber::get_widgets( 'primary-widget-area' );
```

Comment:
Rather than requiring developers to know that they need to update this here when they add a widget, could we move a subset of this function into `theme-widgets.php`? It would be nice to have a `sparkpress_add_widgets_to_context` function in the same file that's clearly related to the widget registration.

Reply:
Also, WordPress has confusing naming (why are widgets called sidebars?), but we don't have to follow those confusing conventions. Can we call these `sidebar_widget` and `footer_widget` so it aligns better with what an admin would see in WordPress menus?

Reply:
@dustin-jw So we would rename and move (most of) this function to `theme-widgets.php`?
```php
function add_to_context( $context ) {
         // don't move this line into theme-widgets
	$context['menu'] = new \Timber\Menu( 'primary-menu' );
	$context['primary_sidebar'] = Timber\Timber::get_widgets( 'primary-widget-area' );
	$context['footer_sidebar'] = Timber\Timber::get_widgets( 'footer-area' );
	return $context;
}
add_filter( 'timber/context', 'add_to_context' );
```

It feels like we could also do something similar with moving `$context['menu']` by creating a function that adds it context in `theme-setup.php`, right? And if we did do that, do I take the `add_to_context` function out of `functions.php` altogether or leave some remnant of it in case future developers want to add things to it later?

Reply:
I think if we left the menu part in `add_to_context` and left it in `functions.php`, that would be most useful for future devs, like it gives them an example of how to globally add something to `$context` in a place that they'll probably end up looking at at some point. Other than changing the theme name, I don't think there's much reason for a dev to interact with `theme-setup.php`, so I think that would obscure how the menu data gets into context.